### ISSUE TITLE
database reinitialized unexpectedly

### ISSUE THREAD

**Issue by dmp1ce** [docker-library/mariadb#42]
Using `docker-compose up -d --force-recreate` mariadb will reinitialize the container causing the database to be dropped, deleting all tables.

To reproduce:

1. Use the `docker-compose.yml` below.
2. `docker-compose up -d`
3. Create a table in mariadb.
4. Run `docker-compose up -d --force-recreate` once or twice.
5. Notice the table created will be gone and the database will reinitialize.

Is `--force-recreate` not supported? I have been using this feature for months with no issues until now. I just recently updated to docker 1.10 and docker-compose 1.6.0. I wasn't able to get the container to reinitialize every time, but most of the time it would reinitialize with `--force-recreate`.

Clearly, this is a major issue for anyone relying on mariadb to store data. Luckily I had a backup of the production site where the databases were dropped unexpectedly.

```
$ docker version
Client:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Fri Feb 12 00:52:02 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Fri Feb 12 00:52:02 2016
 OS/Arch:      linux/amd64
```

```
$ docker-compose version
docker-compose version 1.6.0, build unknown
docker-py version: 1.7.0
CPython version: 3.5.1
OpenSSL version: OpenSSL 1.0.2f  28 Jan 2016
```

Here is my `docker-compose.yml`
```
mariadb:
  image: mariadb
  environment:
    MYSQL_ROOT_PASSWORD: root_password
    MYSQL_USER: app
    MYSQL_PASSWORD: user_password
    MYSQL_DATABASE: app
    TERM: dumb
```

log
```
Attaching to mariadbinit_mariadb_1
mariadb_1 | Initializing database
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] /usr/sbin/mysqld (mysqld 10.1.11-MariaDB-1~jessie) starting as process 56 ...
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Using mutexes to ref count buffer pool pages
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: The InnoDB memory heap is disabled
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Memory barrier is not used
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Compressed tables use zlib 1.2.8
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Using Linux native AIO
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Using SSE crc32 instructions
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Initializing buffer pool, size = 256.0M
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: The first specified data file ./ibdata1 did not exist: a new database to be created!
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Setting file ./ibdata1 size to 12 MB
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Database physically writes the file full: wait...
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Setting log file ./ib_logfile101 size to 48 MB
mariadb_1 | 2016-02-14 16:27:44 139670445746112 [Note] InnoDB: Setting log file ./ib_logfile1 size to 48 MB
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: Renaming log file ./ib_logfile101 to ./ib_logfile0
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Warning] InnoDB: New log files created, LSN=45883
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: Doublewrite buffer not found: creating new
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: Doublewrite buffer created
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: 128 rollback segment(s) are active.
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Warning] InnoDB: Creating foreign key constraint system tables.
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: Foreign key constraint system tables created
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: Creating tablespace and datafile system tables.
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: Tablespace and datafile system tables created.
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB: Waiting for purge to start
mariadb_1 | 2016-02-14 16:27:45 139670445746112 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.26-76.0 started; log sequence number 0
mariadb_1 | 2016-02-14 16:27:45 139669652109056 [Note] InnoDB: Dumping buffer pool(s) not yet started
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] /usr/sbin/mysqld (mysqld 10.1.11-MariaDB-1~jessie) starting as process 84 ...
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Using mutexes to ref count buffer pool pages
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: The InnoDB memory heap is disabled
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Memory barrier is not used
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Compressed tables use zlib 1.2.8
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Using Linux native AIO
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Using SSE crc32 instructions
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Initializing buffer pool, size = 256.0M
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Highest supported file format is Barracuda.
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: 128 rollback segment(s) are active.
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB: Waiting for purge to start
mariadb_1 | 2016-02-14 16:27:47 140660426930112 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.26-76.0 started; log sequence number 1616799
mariadb_1 | 2016-02-14 16:27:48 140659642042112 [Note] InnoDB: Dumping buffer pool(s) not yet started
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] /usr/sbin/mysqld (mysqld 10.1.11-MariaDB-1~jessie) starting as process 113 ...
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Using mutexes to ref count buffer pool pages
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: The InnoDB memory heap is disabled
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Memory barrier is not used
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Compressed tables use zlib 1.2.8
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Using Linux native AIO
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Using SSE crc32 instructions
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Initializing buffer pool, size = 256.0M
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Highest supported file format is Barracuda.
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: 128 rollback segment(s) are active.
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB: Waiting for purge to start
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.26-76.0 started; log sequence number 1616809
mariadb_1 | 2016-02-14 16:27:50 140554620864256 [Note] InnoDB: Dumping buffer pool(s) not yet started
mariadb_1 | ERROR: 1049  Unknown database 'test'
mariadb_1 | 2016-02-14 16:27:50 140555404093376 [ERROR] Aborting
mariadb_1 | 
mariadb_1 | 
mariadb_1 | WARNING: OPENGIS REQUIRED SP-S WERE NOT COMPLETELY INSTALLED!
mariadb_1 | GIS extentions might not work properly.
mariadb_1 | 
mariadb_1 | PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
mariadb_1 | To do so, start the server, then issue the following commands:
mariadb_1 | 
mariadb_1 | '/usr/bin/mysqladmin' -u root password 'new-password'
mariadb_1 | '/usr/bin/mysqladmin' -u root -h  password 'new-password'
mariadb_1 | 
mariadb_1 | Alternatively you can run:
mariadb_1 | '/usr/bin/mysql_secure_installation'
mariadb_1 | 
mariadb_1 | which will also give you the option of removing the test
mariadb_1 | databases and anonymous user created by default.  This is
mariadb_1 | strongly recommended for production servers.
mariadb_1 | 
mariadb_1 | See the MariaDB Knowledgebase at http://mariadb.com/kb or the
mariadb_1 | MySQL manual for more instructions.
mariadb_1 | 
mariadb_1 | Please report any problems at http://mariadb.org/jira
mariadb_1 | 
mariadb_1 | The latest information about MariaDB is available at http://mariadb.org/.
mariadb_1 | You can find additional information about the MySQL part at:
mariadb_1 | http://dev.mysql.com
mariadb_1 | Support MariaDB development by buying support/new features from MariaDB
mariadb_1 | Corporation Ab. You can contact us about this at sales@mariadb.com.
mariadb_1 | Alternatively consider joining our community based development effort:
mariadb_1 | http://mariadb.com/kb/en/contributing-to-the-mariadb-project/
mariadb_1 | 
mariadb_1 | Database initialized
mariadb_1 | MySQL init process in progress...
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] mysqld (mysqld 10.1.11-MariaDB-1~jessie) starting as process 140 ...
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Using mutexes to ref count buffer pool pages
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: The InnoDB memory heap is disabled
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Memory barrier is not used
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Compressed tables use zlib 1.2.8
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Using Linux native AIO
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Using SSE crc32 instructions
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Initializing buffer pool, size = 256.0M
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Highest supported file format is Barracuda.
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: 128 rollback segment(s) are active.
mariadb_1 | 2016-02-14 16:27:52 140311906154432 [Note] InnoDB: Waiting for purge to start
mariadb_1 | 2016-02-14 16:27:53 140311906154432 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.26-76.0 started; log sequence number 1616819
mariadb_1 | 2016-02-14 16:27:53 140311116351232 [Note] InnoDB: Dumping buffer pool(s) not yet started
mariadb_1 | 2016-02-14 16:27:53 140311906154432 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1 | 2016-02-14 16:27:53 140311906154432 [Warning] 'user' entry 'root@884b4f5d628e' ignored in --skip-name-resolve mode.
mariadb_1 | 2016-02-14 16:27:53 140311906154432 [Warning] 'proxies_priv' entry '@% root@884b4f5d628e' ignored in --skip-name-resolve mode.
mariadb_1 | 2016-02-14 16:27:53 140311906154432 [Note] mysqld: ready for connections.
mariadb_1 | Version: '10.1.11-MariaDB-1~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 0  mariadb.org binary distribution
mariadb_1 | Warning: Unable to load '/usr/share/zoneinfo/leap-seconds.list' as time zone. Skipping it.
mariadb_1 | 2016-02-14 16:27:54 140311905590016 [Warning] 'proxies_priv' entry '@% root@884b4f5d628e' ignored in --skip-name-resolve mode.
mariadb_1 | 2016-02-14 16:27:54 140311905590016 [Warning] 'proxies_priv' entry '@% root@884b4f5d628e' ignored in --skip-name-resolve mode.
mariadb_1 | 
mariadb_1 | /docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
mariadb_1 | 
mariadb_1 | 2016-02-14 16:27:54 140311905286912 [Note] mysqld: Normal shutdown
mariadb_1 | 
mariadb_1 | 2016-02-14 16:27:54 140311905286912 [Note] Event Scheduler: Purging the queue. 0 events
mariadb_1 | 2016-02-14 16:27:54 140311099565824 [Note] InnoDB: FTS optimize thread exiting.
mariadb_1 | 2016-02-14 16:27:54 140311905286912 [Note] InnoDB: Starting shutdown...
mariadb_1 | 2016-02-14 16:27:57 140311905286912 [Note] InnoDB: Shutdown completed; log sequence number 1616829
mariadb_1 | 2016-02-14 16:27:57 140311905286912 [Note] mysqld: Shutdown complete
mariadb_1 | 
mariadb_1 | 
mariadb_1 | MySQL init process done. Ready for start up.
mariadb_1 | 
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] mysqld (mysqld 10.1.11-MariaDB-1~jessie) starting as process 1 ...
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Using mutexes to ref count buffer pool pages
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: The InnoDB memory heap is disabled
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Memory barrier is not used
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Compressed tables use zlib 1.2.8
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Using Linux native AIO
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Using SSE crc32 instructions
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Initializing buffer pool, size = 256.0M
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Highest supported file format is Barracuda.
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: 128 rollback segment(s) are active.
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB: Waiting for purge to start
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.26-76.0 started; log sequence number 1616829
mariadb_1 | 2016-02-14 16:27:57 139960468342528 [Note] InnoDB: Dumping buffer pool(s) not yet started
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] Server socket created on IP: '::'.
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Warning] 'proxies_priv' entry '@% root@884b4f5d628e' ignored in --skip-name-resolve mode.
mariadb_1 | 2016-02-14 16:27:57 139961255913408 [Note] mysqld: ready for connections.
mariadb_1 | Version: '10.1.11-MariaDB-1~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
```

**Comment by dmp1ce** [comment]
Nevermind, the passenger service seems to be re-initializing the database on restart.

**Comment by yosifkit** [comment]
This was fixed in [compose 1.6.1](https://github.com/docker/compose/releases/tag/1.6.1).
